services/provider/server/server.go

```diff
@@ -204,3 +224,120 @@ func Start(port int, providerServer *OCSProviderServer, opts []grpc.ServerOption
 		klog.Fatalf("failed to start gRPC server: %v", err)
 	}
 }
+func (s *OCSProviderServer) getJSONConnectionDetails(consumerResource *ocsv1alpha1.StorageConsumer) ([]externalSC.ExternalResource, error) {
+	var extR []externalSC.ExternalResource
+	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt)
+	defer stop()
+
+	// Configmap with mon endpoints
+	configmap, err := s.consumerAPI.GetConfigMap(ctx, monConfigMap, s.namespace)
+	if err != nil {
+		return []externalSC.ExternalResource{}, fmt.Errorf("failed to get %s configMap. %v", monConfigMap, err)
+	}
+
+	// Get address of first mon from the monConfigMap configmap
+	cmData := strings.Split(configmap.Data["data"], ",")
+	// Address of first mon
+	monAddr := cmData[0]
+	if monAddr == "" {
+		return []externalSC.ExternalResource{}, fmt.Errorf("configmap %s data is empty", monConfigMap)
+	}
+
+	data := map[string]string{
+		"data":     monAddr,
+		"maxMonId": "0",
+		"mapping":  "{}",
+	}
+	name := monConfigMap
+	kind := "ConfigMap"
+
+	extR = append(extR, externalSC.ExternalResource{Name: name, Kind: kind, Data: data})
+
+	// Secret storing cluster mon.admin key, fsid and name
+	scMon, err := s.consumerAPI.GetSecret(ctx, monSecret, s.namespace)
+	if err != nil {
+		return []externalSC.ExternalResource{}, fmt.Errorf("failed to get %s secret. %v", monSecret, err)
+	}
+	if string(scMon.Data["fsid"]) == "" {
+		return []externalSC.ExternalResource{}, fmt.Errorf("secret %s data fsid is empty", monSecret)
+	}
+
+	data = map[string]string{
+		"fsid":         string(scMon.Data["fsid"]),
+		"mon-secret":   "mon-secret",
```

Comment:
Should the `mon-secret` be set to the text "mon-secret" like this?

Reply:
yes, check [this](https://github.com/rook/rook/blob/master/deploy/examples/create-external-cluster-resources.py#L912-L915).